packages/flutter/test/cupertino/expansion_tile_test.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// reduced-test-set:
+//   This file is run as part of a reduced test set in CI on Mac and Windows
+//   machines.
+@Tags(<String>['reduced-test-set'])
+library;
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Toggles expansion on tap', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const CupertinoApp(
+        home: CupertinoPageScaffold(
+          child: CupertinoExpansionTile(title: Text('Title'), child: Text('Content')),
+        ),
+      ),
+    );
+
+    expect(find.text('Content'), findsNothing);
+
+    await tester.tap(find.text('Title'));
+    await tester.pump();
+    // The child animating its height and a clone fading in.
+    expect(find.text('Content'), findsNWidgets(2));
+
+    await tester.tap(find.text('Title'));
+    await tester.pump();
+    expect(find.text('Content'), findsNothing);
+  });
+
+  testWidgets('Can be controlled by ExpansibleController', (WidgetTester tester) async {
+    final ExpansibleController controller = ExpansibleController();
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: CupertinoPageScaffold(
+          child: CupertinoExpansionTile(
+            controller: controller,
+            title: const Text('Title'),
+            child: const Text('Content'),
+          ),
+        ),
+      ),
+    );
+
+    expect(controller.isExpanded, isFalse);
+    expect(find.text('Content'), findsNothing);
+
+    controller.expand();
+    await tester.pump();
+    expect(controller.isExpanded, isTrue);
+    expect(find.text('Content'), findsOneWidget);
+
+    controller.collapse();
+    await tester.pump();
+    expect(controller.isExpanded, isFalse);
+    expect(find.text('Content'), findsNothing);
+
+    controller.dispose();
+  });
+
+  testWidgets('Nested expansion tile', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const CupertinoApp(
+        home: CupertinoPageScaffold(
+          child: CupertinoExpansionTile(
+            title: Text('Outer'),
+            child: CupertinoExpansionTile(title: Text('Inner'), child: Text('Content')),
+          ),
+        ),
+      ),
+    );
+
+    expect(find.text('Content'), findsNothing);
+
+    await tester.tap(find.text('Outer'));
+    await tester.pumpAndSettle();
```

Comment:
Nit: You could make this test a bit more strict by making _kAnimationDuration visibleForTesting and pumping for exactly that duration.

Reply:
[Related styleguide rule](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-pumpandsettle).